BaseAdresseNationale/ban-plateforme
# [PRIO] Active state and historic mechanism

# Context 

We need to be able to deactivate and re-activate an address, a common toponym or a district. We also need to have in the historic a trace of a deactivation and reactivation. 

# Enchancement 

This PRs aims to :
- Add a "isActive" column in all tables (address, common_toponym and district)
- Modify all the "destroy" fonctions to "update" with the flag : isActive = false
- Modify all the "updates" functions to "update" with the flag : isActive = true (to be able to re-activate an item with a simple update (so that a district can simply re-activate an item simply by adding again the line in the bal)
- Rebuild the historic mechanism

# How to test 
I. Initialization : 

If using docker and docker-compose : 
- Start the following command : 
`docker compose up --build -d` <= migrations are done automatically when starting the containers

If not using docker : 
- Start the migrations with the following command :
`yarn run migrate:up`

- Start the ban-pateforme : 
`yarn worker`
`yarn worker:dev`

II. Testing process
In both cases : 
- Create a district with an API call to `POST /district/`.
- Send a BAL using id-fix command `initBALIntoBAN`.
- Check your local database tables on 'address', 'common_toponym' and 'district' and especially the new column.'isActive' that should be `true` for all lines.
- Modify the BAL you sent by deleting one line for exemple.
- Check in your DB that the corresponding address or common toponym has its flag "isActive" to false
- Check that the final result in Mongo DB does not have the deleted address or common toponym.
